CHANGELOG.md

```diff
@@ -10,6 +10,16 @@ to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ### Breaking changes
 
+### Deprecations
+
+### New features
+
+### Fixed bugs
+
+## [0.24.0] - 2024-12-04
```

Comment:
I wonder if we should put the new dynamic completions support in a "Highlights" section or something, since it's been a huge ask for so long. Either way, LGTM.

Reply:
Makes sense, but I think I remember reading that dynamic completions only work well with Fish so far, so we'll probably want to fix the bugs (in jj, clap, or elsewhere) and test it a bit more first.